crates/miden-lib/asm/miden/auth/rpo_falcon512.masm

```diff
@@ -23,55 +23,46 @@ const.PUBLIC_KEY_SLOT=0
 #! included to commit to the transaction creator's intended reference block of the transaction
 #! which determines the fee parameters and therefore the fee amount that is deducted.
 #!
-#! Inputs:  [AUTH_ARGS, pad(12)]
-#! Outputs: [pad(16)]
+#! Inputs:  []
+#! Outputs: []
 #!
-#! Invocation: call
-export.auth__tx_rpo_falcon512
-    # drop auth args
-    dropw
-    # => [pad(16)]
-
+#! Invocation: exec
+export.authenticate_transaction
     # Fetch public key from storage.
     # ---------------------------------------------------------------------------------------------
-
-    # Get public key from account storage at pos 0 and verify signature
     push.PUBLIC_KEY_SLOT exec.account::get_item
+    # => [PUB_KEY]
```

Comment:
An alternative could be to pass `PUB_KEY` as an input parameter into this procedure. This decouples it from the assumptions about the storage. I don't have a specific reason for this in mind, but it would feel a bit more self-contained this way.

Reply:
I like that suggestion 👍🏼